From 0509250af66125bc80b166cbbd82a3fe76c6f581 Mon Sep 17 00:00:00 2001
From: "Tekriwal, Tanuj" <tanuj.tekriwal@intel.com>
Date: Sun, 27 Aug 2023 15:09:49 +0000
Subject: [PATCH] Upgrading API to 34 for android U

This patch:
1. Updates the API to 34 and target to 8
2. Removes the unused graphics composer hal 2.4
3. Updates the ndk version to 34
4. Updates sepolicy version to 34

Tracked-On: OAM-110604
Signed-off-by: Tanuj Tekriwal <tanuj.tekriwal@intel.com>
---
 groups/device-specific/caas/caas.mk             |  2 +-
 .../caas/compatibility_matrix.xml               |  6 +++---
 .../device-specific/caas/framework_manifest.xml | 15 +++------------
 groups/device-specific/caas/manifest.xml        | 17 ++++-------------
 groups/device-specific/caas/product.mk          |  1 -
 5 files changed, 11 insertions(+), 30 deletions(-)

diff --git a/groups/device-specific/caas/caas.mk b/groups/device-specific/caas/caas.mk
index 7a1b952..c4139fe 100644
--- a/groups/device-specific/caas/caas.mk
+++ b/groups/device-specific/caas/caas.mk
@@ -71,7 +71,7 @@ else
 PRODUCT_SET_DEBUGFS_RESTRICTIONS := true
 endif
 {{^ota-update}}
-PRODUCT_SHIPPING_API_LEVEL := 33
+PRODUCT_SHIPPING_API_LEVEL := 34
 PRODUCT_OTA_ENFORCE_VINTF_KERNEL_REQUIREMENTS := true
 {{/ota-update}}
 {{#ota-update}}
diff --git a/groups/device-specific/caas/compatibility_matrix.xml b/groups/device-specific/caas/compatibility_matrix.xml
index 7f2dab0..d941dc9 100644
--- a/groups/device-specific/caas/compatibility_matrix.xml
+++ b/groups/device-specific/caas/compatibility_matrix.xml
@@ -39,7 +39,7 @@
             <instance>default</instance>
         </interface>
     </hal>
-    <!--vendor-ndk>
-        <version>33</version>
-    </vendor-ndk-->
+    <vendor-ndk>
+        <version>34</version>
+    </vendor-ndk>
 </compatibility-matrix>
diff --git a/groups/device-specific/caas/framework_manifest.xml b/groups/device-specific/caas/framework_manifest.xml
index 14d4d55..5f22203 100644
--- a/groups/device-specific/caas/framework_manifest.xml
+++ b/groups/device-specific/caas/framework_manifest.xml
@@ -97,15 +97,6 @@
             <instance>default</instance>
         </interface>
     </hal>
-    <hal format="hidl">
-        <name>android.hardware.graphics.composer</name>
-        <transport>hwbinder</transport>
-        <version>2.4</version>
-        <interface>
-            <name>IComposer</name>
-            <instance>default</instance>
-        </interface>
-    </hal>
     <hal format="hidl">
         <name>android.hardware.media.omx</name>
         <transport>hwbinder</transport>
@@ -146,7 +137,7 @@
             <instance>default</instance>
         </interface>
     </hal>
-    <!--vendor-ndk>
-        <version>33</version>
-    </vendor-ndk-->
+    <vendor-ndk>
+        <version>34</version>
+    </vendor-ndk>
 </manifest>
diff --git a/groups/device-specific/caas/manifest.xml b/groups/device-specific/caas/manifest.xml
index ae957fb..4877255 100644
--- a/groups/device-specific/caas/manifest.xml
+++ b/groups/device-specific/caas/manifest.xml
@@ -1,5 +1,5 @@
 {{^ota-update}}
-<manifest version="1.0" type="device" target-level="7">
+<manifest version="1.0" type="device" target-level="8">
 {{/ota-update}}
 {{#ota-update}}
 <manifest version="1.0" type="device">
@@ -27,15 +27,6 @@
             <instance>default</instance>
         </interface>
     </hal>
-    <hal format="hidl">
-        <name>android.hardware.graphics.composer</name>
-        <transport>hwbinder</transport>
-        <version>2.4</version>
-        <interface>
-            <name>IComposer</name>
-            <instance>default</instance>
-        </interface>
-    </hal>
     <hal format="aidl">
         <name>android.hardware.graphics.composer3</name>
         <version>2</version>
@@ -81,7 +72,7 @@
         </interface>
     </hal>
     <!--<kernel target-level="5"/>-->
-    <!--sepolicy>
-        <version>33.0</version>
-    </sepolicy-->
+    <sepolicy>
+        <version>34.0</version>
+    </sepolicy>
 </manifest>
diff --git a/groups/device-specific/caas/product.mk b/groups/device-specific/caas/product.mk
index eb717cd..65296ad 100755
--- a/groups/device-specific/caas/product.mk
+++ b/groups/device-specific/caas/product.mk
@@ -29,7 +29,6 @@ PRODUCT_PACKAGES +=  \
                     android.hardware.graphics.mapper@4.0-impl.minigbm_intel \
                     android.hardware.graphics.allocator-service.minigbm \
                     android.hardware.renderscript@1.0-impl \
-                    android.hardware.graphics.composer@2.4-service \
                     android.hardware.identity \
                     android.hardware.identity-service.example \
                     android.hardware.graphics.composer3-service.intel \
-- 
2.17.1

